MetasysServices.Tests/MetasysClientTests.cs

```diff
@@ -2412,12 +2414,189 @@ public void TestSendAsyncWithInvalidAbsoluteUrlThrowsException()
+
+            Assert.AreEqual(1, data.id);
+            Assert.AreEqual("Metasys", data.name);
+        }
+
+        [Test]
+        public async Task TestSendAsyncCanGetJsonDynamicList()
+        {
+            httpTest.RespondWithJson(
+                new[] {
+                    new { id = 1, name = "Metasys" },
+                    new { id = 2, name = "Client" }
+                }
+            );
+
+            var httpRequest = new HttpRequestMessage(HttpMethod.Get, "https://hostname/api/v5/getJsonDynamicList");
+            var data = await client.SendAsync(httpRequest).ReceiveJsonList();
+
+            Assert.AreEqual(1, data[0].id);
+            Assert.AreEqual("Metasys", data[0].name);
+            Assert.AreEqual(2, data[1].id);
+            Assert.AreEqual("Client", data[1].name);
+        }
+
+        [Test]
+        public async Task TestSendAsyncCanGetString()
+        {
+            httpTest.RespondWith("Metasys Client");
+
+            var httpRequest = new HttpRequestMessage(HttpMethod.Get, "https://hostname/api/v5/getString");
+            var data = await client.SendAsync(httpRequest).ReceiveString();
+
+            Assert.AreEqual("Metasys Client", data);
+        }
+
+        [Test]
+        public async Task TestSendAsyncCanGetStream()
+        {
+            httpTest.RespondWith("Metasys Client");
+
+            var httpRequest = new HttpRequestMessage(HttpMethod.Get, "https://hostname/api/v5/getStream");
+            var data = await client.SendAsync(httpRequest).ReceiveStream();
+
+            Assert.AreEqual(new MemoryStream(Encoding.UTF8.GetBytes("Metasys Client")), data);
+        }
+
+        [Test]
+        public async Task TestSendAsyncCanGetBytes()
+        {
+            httpTest.RespondWith("Metasys Client");
+
+            var httpRequest = new HttpRequestMessage(HttpMethod.Get, "https://hostname/api/v5/getBytes");
+            var data = await client.SendAsync(httpRequest).ReceiveBytes();
+
+            Assert.AreEqual(Encoding.UTF8.GetBytes("Metasys Client"), data);
+        }
+
+        [Test]
+        public async Task TestSendAsyncCanMakePutRequest()
+        {
+            var httpRequest = new HttpRequestMessage(HttpMethod.Put, "https://hostname/api/v5/put")
+            {
+                Content = JsonContent.Create(new { id = 1, name = "Metasys" })
+            };
+            
+            var response = await client.SendAsync(httpRequest);
+
+            Assert.AreEqual(HttpStatusCode.OK, response.StatusCode);
+        }
+
+        [Test]
+        public async Task TestSendAsyncCanMakePostRequest()
+        {
+            var httpRequest = new HttpRequestMessage(HttpMethod.Post, "https://hostname/api/v5/post")
+            {
+                Content = JsonContent.Create(new { id = 1, name = "Metasys" })
+            };
+
+            var response = await client.SendAsync(httpRequest);
+
+            Assert.AreEqual(HttpStatusCode.OK, response.StatusCode);
```

Comment:
I would expect that you'd assert the content was actually in the request.

Reply:
done